src/Security/Voter/AdminRestrictionVoter.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\Security\Voter;
+
+use App\Entity\Participation;
+use App\Entity\Talk;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class AdminRestrictionVoter extends Voter
+{
+    protected function supports($attribute, $subject): bool
+    {
+        return \in_array($attribute, ['OWNER'])
+            && ($subject instanceof Talk || $subject instanceof Participation);
+    }
+
+    /** @phpstan-ignore-next-line */
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token): bool
+    {
+        $user = $token->getUser();
+        if (!$user instanceof UserInterface) {
+            return false;
+        }
+
+        if ($subject instanceof Talk) {
+            if (\in_array($user, $subject->getUniqueUsers())) {
+                return true;
+            }
```

Comment:
This code could be enchanced: in `$subject->getUniqueUsers()` you foreach all the record, the create a buffer. This consumer many memory for nothing. Let's move the foreach here a break as soon as the users is concerned. or you can move the code in the subject `return $suject->isUserAMember($user)`